extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/automatic_speech_recognition.py

```diff
@@ -163,21 +163,24 @@ def validate_and_retrieve_audio_from_attachments(prompt: Prompt) -> list[str]:
     - operation fails for any reason
     """
 
-    if not hasattr(prompt.input, "attachments") or len(prompt.input.attachments) == 0:
+    if not isinstance(prompt.input, PromptInput):
+        raise ValueError(f"Prompt input is of type {type(prompt.input) }. Please specify a PromptInput with attachments for prompt {prompt.name}.")
+    
+    if prompt.input.attachments is None or len(prompt.input.attachments) == 0:
         raise ValueError(f"No attachments found in input for prompt {prompt.name}. Please add an audio attachment to the prompt input.")
 
-    audio_uris: list[str] = []
+    audio_inputs: list[str] = []
 
     for i, attachment in enumerate(prompt.input.attachments):
         validate_attachment_type_is_audio(attachment)
 
-        if not isinstance(attachment.data, str):
-            # See todo above, but for now only support uri's
-            raise ValueError(f"Attachment #{i} data is not a uri. Please specify a uri for the audio attachment in prompt {prompt.name}.")
+        if not isinstance(attachment.data, AttachmentDataWithStringValue):
```

Comment:
Do we need to have backwards-compatibility here for potential existing configs with the older format?

Reply:
> Do we need to have backwards-compatibility here for potential existing configs with the older format?

I believe gradio example was the only config that used it. I think also what @saqadri meant by backwards-compatibility was that the AIConfig itself should be able to be loaded with `AIConfigRuntime.load()` if it doesn't use `AttachmentDataWithStringValue`

Reply:
If that's not the case though we can allow arbitrary attachment data and let the model error as needed, but not sure if thats what we want

Reply:
discussed offline with @saqadri, lets ship this, mark this as breaking change in changelog to keep anyone aware because this is technically a breaking change but doubt on who is using it but we want to inform anyone who is using it just in case.